<?xml version="1.0"?>
<launch>
  <!-- our xacro/urdf model -->
  <arg name="model" value="$(find pendulum)/urdf/pendulum.xacro"/>
  <arg name="tracker_model" value="$(find pendulum)/urdf/tracker.xacro"/>

  <!-- load our urdf model (string) into global parameter called "robot_description"-->
  <!-- xacro parses xacro files and returns a urdf file -->
  <param name="robot_description"
         command="$(find xacro)/xacro $(arg model)" />

  <param name="tracker_description"
         command="$(find xacro)/xacro $(arg tracker_model)" />

  <!-- We use empty_world.launch to start gazebo, changing only the name of the world to be launched -->
  <include file="$(find pendulum)/launch/empty_world.launch">
    <arg name="debug" value="false" />
    <arg name="gui" value="true" />
    <arg name="paused" value="true"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="headless" value="false"/>
  </include>

  <!-- Spawn our robot using urdf_spawner node -->
  <node name="urdf_spawner"
        pkg="gazebo_ros"
        type="spawn_model"
        respawn="false"
        output="screen"
        args="-z 0
              -urdf
              -model robot
              -param robot_description
              -J robot::arm_spring '-0.05918'" />
              <!-- -J robot::base_free_theta 0.8"/> --><!--
              -R 1.8 -->

  <node name="tracker_spawner"
        pkg="gazebo_ros"
        type="spawn_model"
        respawn="false"
        output="screen"
        args="-urdf
              -model tracker
              -param tracker_description"/>

  <!-- load multiple values stored in .yaml file into global parameters -->
  <!-- parameters store motor controller configurations -->
  <rosparam command="load"
            file="$(find pendulum)/config/joints.yaml" />

  <!-- Base link contact listener -->
  <node name="base_contact"
        pkg="pendulum"
        respawn="true"
        type="base_contact"
   />

  <!-- Spawn controllers -->
  <node name="pend_main_ctrl"
        pkg="pendulum"
        respawn="true"
        type="my_controller"
    />

  <node name="prism_ctrl"
      pkg="pendulum"
      respawn="true"
      type="prism_ctrl"
  />

  <node name="spring_emulator"
        pkg="pendulum"
        respawn="true"
        type="spring_emulator"
  />

  <node name="bot_controller_spawner"
        pkg="controller_manager"
        type="spawner"
        respawn="true"
        args="joint_state_controller
              flywheel_controller
              arm_spring
              arm_controller" />
</launch>


